### PR TITLE
Order, OrderDao, OrderService classes implemented

### PR DESCRIPTION
**Pull request by semyonich**

### CODE REVIEW

**Review by AndreiYazadzhi — APPROVED** [review]

File: src/main/java/com/cinema/Application.java [hunk 1]
```diff
@@ -66,6 +69,16 @@ public static void main(String[] args) {
         shoppingCartService.addSession(movieSession2, user2);
         shoppingCartService.addSession(movieSession2, user1);
         ShoppingCart user1ShoppingCart = shoppingCartService.getByUser(user1);
-        shoppingCartService.clear(user1ShoppingCart);
+        ShoppingCart user2ShoppingCart = shoppingCartService.getByUser(user2);
+        //shoppingCartService.clear(user1ShoppingCart);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/Application.java]:
```suggestion
```

File: src/main/java/com/cinema/dao/impl/OrderDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import com.cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public List<Order> getOrdersHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM Order o "
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/dao/impl/OrderDaoImpl.java]: https://mateacademy.slack.com/archives/G01EHB8CWAE/p1612455330236200

File: src/main/java/com/cinema/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        List<Ticket> tickets = shoppingCart.getTickets();
+        User user = shoppingCart.getUser();
+        Order order = new Order();
+        order.setUser(user);
+        order.setTickets(tickets);
+        order.setOrderDate(LocalDateTime.now());
+        orderDao.add(order);
+        shoppingCart.setTickets(new ArrayList<>());
+        shoppingCartDao.update(shoppingCart);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/service/impl/OrderServiceImpl.java]: It's okay to user one service inside other. Let's use ShoppingCartService.clear() here/

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/Application.java (hunk 1 above)
Comment (semyonich) [reply src/main/java/com/cinema/Application.java]: done

File: src/main/java/com/cinema/dao/impl/OrderDaoImpl.java (hunk 2 above)
Comment (semyonich) [reply src/main/java/com/cinema/dao/impl/OrderDaoImpl.java]: fixed

File: src/main/java/com/cinema/service/impl/OrderServiceImpl.java (hunk 3 above)
Comment (semyonich) [reply src/main/java/com/cinema/service/impl/OrderServiceImpl.java]: done

File: src/main/java/com/cinema/service/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -30,7 +30,7 @@ public Order completeOrder(ShoppingCart shoppingCart) {
         order.setOrderDate(LocalDateTime.now());
         orderDao.add(order);
         shoppingCart.setTickets(new ArrayList<>());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/cinema/service/impl/OrderServiceImpl.java:32]: do we need this?

**Review by dmytro-hryhoruk — APPROVED** [review]

File: src/main/java/com/cinema/service/impl/OrderServiceImpl.java (hunk 4 above)
Comment (semyonich) [reply src/main/java/com/cinema/service/impl/OrderServiceImpl.java:32]: yes, without it calling .clear() deletes all tickets from orders

**Pull request merged**